### PR TITLE
Add dpnp.linalg.eigh() function

### LINKED ISSUE #1344: Support PCA dpnp implementation on dpbench
The [dpnp implementation of the PCA algorithm](https://github.com/adarshyoga/dpbench/blob/main/dpbench/benchmarks/pca/pca_dpnp.py) has several calls that are currently not supported in dpnp - dpnp.mean(axis=0), dpnp.linalg.eigh, etc. These functions need to supported in dpnp to run PCA workload in dpbench.

How to reproduce:
1) Follow [instructions ](https://github.com/adarshyoga/dpbench/blob/main/README.md)to setup dpbench
2) Run PCA - `python -c "import dpbench; dpbench.run_benchmark(\"pca\")`

### PR DESCRIPTION
**Pull request by antonwolfy**
The PR is about to close #1344, since it covers the last functional gap of `pca` benchmark.

PR adds a missing `dpnp.linalg.eigh()` function to calculate the eigenvalues and eigenvectors of a complex Hermitian (conjugate symmetric) or a real symmetric input matrix. OneMKL LAPACK library is used for that.
The implementation is written as a pybind11 extension above required LAPACK functions.

This is a base PR which assumes to be extended later with remaining functions from OneMKL LAPACK library, once new functionality required.

Numbers for a laptop with Iris Xe:
```
import dpnp, numpy

sh = (2024, 2024)
a = numpy.random.uniform(-1, 1, sh) + 1.j * numpy.random.uniform(-1, 1, sh)
a = a.astype(numpy.complex64)
x = numpy.matrix(a)

nx = x.getH()
%timeit _, _ = numpy.linalg.eigh(nx)
Out: 2.21 s ± 199 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

dx = dpnp.array(nx, device='level_zero:gpu')
%timeit _, _ = dpnp.linalg.eigh(dx)
Out: 858 ms ± 25.2 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

dx = dpnp.array(nx, device='opencl:gpu')
%timeit _, _ = dpnp.linalg.eigh(dx)
Out: 849 ms ± 37.3 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

dx = dpnp.array(nx, device='opencl:cpu')
%timeit _, _ = dpnp.linalg.eigh(dx)
Out: 841 ms ± 47.8 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

nx.size, dx.size
Out: (4096576, 4096576)
```

- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [x] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # CMake build and local install directory
 _skbuild
+build
```
Comment (oleksandr-pavlyk) [inline .gitignore:3]: Curiosity: why do you find this necessary?

Comment (antonwolfy) [reply .gitignore:3]: It's a folder generated during the build, isn't it?

File: dpnp/backend/extensions/lapack/heevd.cpp [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    auto const& dpctl_capi = dpctl::detail::dpctl_capi::get();
+
+    sycl::event heevd_ev;
+    std::vector<sycl::event> host_task_events;
+
+    const std::int64_t n = eig_vecs_shape[0];
+    const oneapi::mkl::job jobz_val = static_cast<oneapi::mkl::job>(jobz);
+    const oneapi::mkl::uplo uplo_val = static_cast<oneapi::mkl::uplo>(upper_lower);
+
+    if ((eig_vecs_typenum == dpctl_capi.UAR_CDOUBLE_) && (eig_vals_typenum == dpctl_capi.UAR_DOUBLE_))
```
Comment (oleksandr-pavlyk) [inline dpnp/backend/extensions/lapack/heevd.cpp:185]: You are assuming that both arrays are C-contiguous, I suppose, but the function does not check for that.

File: dpnp/backend/extensions/lapack/heevd.cpp [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//*****************************************************************************
+
+
+#include <pybind11/pybind11.h>
+
+#include "heevd.hpp"
+
+#include "dpnp_utils.hpp"
```
[the remark this answers is not in the document]
Comment (antonwolfy) [reply dpnp/backend/extensions/lapack/heevd.cpp:31]: it is not done yet

File: .gitignore (hunk 1 above)
Comment (oleksandr-pavlyk) [reply .gitignore:3]: It used to, when we used `setuptools`. With transition to `scikit-build` the generated folder is `_skbuild`.

Comment (antonwolfy) [reply .gitignore:3]: Looks I've not cleaned up the env for a long time. I will roll back the change.

File: dpnp/backend/extensions/lapack/heevd.cpp (hunk 2 above)
Comment (antonwolfy) [reply dpnp/backend/extensions/lapack/heevd.cpp:196]: `eig_vecs` is expected to be F-contiguous, while `eig_vals` are C-contiguous. I will add the check.

File: dpnp/backend/extensions/lapack/heevd.cpp [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+namespace ext
+{
+namespace lapack
+{
+
+namespace mkl_lapack = oneapi::mkl::lapack;
+namespace py = pybind11;
+
+template <typename T, typename RealT>
+static inline sycl::event call_heevd(sycl::queue exec_q,
```
[the remark this answers is not in the document]
Comment (antonwolfy) [reply dpnp/backend/extensions/lapack/heevd.cpp]: removed

File: dpnp/dpnp_iface.py [hunk 5]
```diff
@@ -371,3 +373,44 @@ def get_normalized_queue_device(obj=None,
...
+        or :class:`dpctl.tensor.usm_ndarray`.
+
+    Returns
+    -------
+    out : usm_ndarray
+        A dpctl USM ndarray of input array `a`.
+
+    """
+
+    return a.get_array() if isinstance(a, dpnp_array) else a
```
Comment (npolina4) [inline dpnp/dpnp_iface.py:395]: Can you add a check for input argument?

**Review by npolina4 — APPROVED** [review]
Works in pca workload. Thanks!

**Pull request merged**